caproto/threading/client.py

```diff
@@ -1789,6 +1792,110 @@ def __del__(self):
             pass
 
 
+class Batch:
+    """
+    Accumulate requests and then issue them all in batch.
+
+    Parameters
+    ----------
+    timeout : number or None
+        Number of seconds to wait before ignoring late responses. Default
+        is 2.
+
+    Examples
+    --------
+    Read some PVs in batch and stash the readings in a dictionary as they
+    come in.
+
+    >>> readings = {}
+
+    >>> results = {}
+    >>> stash_result = lambda name, response: results.update({name: response.data})
+
+    >>> with Batch() as b:
+    ...     for pv in pvs:
+    ...         b.read(pv, functools.partial(stash_result, pv.name))
```

Comment:
I think it's good as-is, but thinking of alternatives: what if `read` also gave you a `Future` of sorts for getting the result of each request? A dictionary? Hmm...

Reply:
I think the asyncio variety should return a Future. Trio doesn't _have_ futures, so not sure about that. I think we should add _something_ once we see how this shakes out in other frameworks.